flight/PiOS/inc/pios_com.h

```diff
@@ -55,6 +55,7 @@ extern int32_t PIOS_COM_ChangeBaud(uintptr_t com_id, uint32_t baud);
 extern int32_t PIOS_COM_SendCharNonBlocking(uintptr_t com_id, char c);
 extern int32_t PIOS_COM_SendChar(uintptr_t com_id, char c);
 extern int32_t PIOS_COM_SendBufferNonBlocking(uintptr_t com_id, const uint8_t *buffer, uint16_t len);
+extern int32_t PIOS_COM_SendBufferProgressWatch(uintptr_t com_id, const uint8_t *buffer, uint16_t len, uint32_t max_ms);
```

Comment:
With this name I'd expect a progress callback. PIOS_COM_SendBufferAsyncTimeout would make a lot more sense to me (but only if /s/PIOS_COM_SendBufferNonBlocking/PIOS_COM_SendBufferAsync/, so alternately the seriously long PIOS_COM_SendBufferNonBlockingTimeout to minimise diff :/).

Reply:
So, the problem is-- it's blocking, and there's not a strict timeout.  Rather, there's a timeout when there's no progress made for a defined interval.  Indeed, it's the exact same thing as SendBuffer (the blocking call) was when 5000 is passed for the last arg.  Better name very welcome :/

Reply:
Ah, I was looking at the wrong one :P. Sometimes I can see the wins of C++ and overloading would be one here. PIOS_COM_SendBufferTimeout says more about it IMO but... :/ I definitely think ProgressWatch isn't right though, because it provides no way to watch progress.

Reply:
It's not too bad, because I took the existing SendBuffer code and just have it call this with one more arg-- so it's pretty much a default value.

I have StallTimeout now...

Reply:
That's a good name. :+1: